source/lac/scalapack.cc

```diff
@@ -1320,6 +1507,46 @@ void ScaLAPACKMatrix<NumberType>::load_parallel(const char *filename)
 
 
 
+template <typename NumberType>
+void ScaLAPACKMatrix<NumberType>::scale_columns(std::vector<NumberType> &factors)
+{
+  Assert(n_columns==(int)factors.size(),ExcDimensionMismatch(n_columns,factors.size()));
+
+  if (this->grid->mpi_process_is_active)
+    {
+      for (int i=0; i<n_local_rows; ++i)
+        {
+          for (int j=0; j<n_local_columns; ++j)
```

Comment:
you should probably loop over columns first, take value and multiply all row elements.